prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/ConstantResolver.kt

```diff
@@ -130,24 +129,25 @@ class FhirPathCustomResolver : FHIRPathEngine.IEvaluationContext, Logging {
         }
 
         // Evaluate the constant before it is used.
-        val value = if (constantValue.isNullOrBlank()) null
+        val values = if (constantValue.isNullOrBlank()) null
         else {
             val values = FhirPathUtils.evaluate(appContext, appContext.focusResource, appContext.bundle, constantValue)
             if (values.isEmpty()) {
                 null
-            } else if (values.size != 1) {
-                // TODO: Support multiple values with the updated function return type and remove our custom solution
-                throw SchemaException("Constant $name must resolve to one value, but had ${values.size}.")
             } else {
-                logger.trace("Evaluated FHIR Path constant $name to: ${values[0]}")
-                // Convert string constants that are whole integers to Integer type to facilitate math operations
-                if (values[0] is StringType && StringUtils.isNumeric(values[0].primitiveValue())) {
-                    IntegerType(values[0].primitiveValue())
-                } else values[0]
+                logger.trace("Evaluated FHIR Path constant $name to: $values")
+                val cleanedValues = mutableListOf<Base>()
```

Comment:
If anyone has a nicer way to do this chunk of code in Kotlin let me know

Reply:
I don't have a better way to do this but I do think its worth adding this comment `// Convert string constants that are whole integers to Integer type to facilitate math operations` back to explain why you're doing this.

Reply:
I think you could do something like 
```
val cleanedValues = values.map {
    if (it is StringType && ...) {
        IntegerType(it.primitiveValue()
    } else {
        it
    }
}
```
Also at that point, you probably don't need to even save it to a new val `cleanedValues`, you could just leave that expression.

Reply:
Looks good in latest commit, marking this conversation resolved.